### PR TITLE
Integrate CAB Forum Network Security Requirements

### PR DESCRIPTION
**Pull request by weirdscience**
Integrate the CAB Forum Network Security Requirements to fix issue #96.

### CODE REVIEW

**Comment by weirdscience** [comment]
Not completely done yet, but finished with most of the Section 6 changes.

File: certificate-policy.md [hunk 1]
```diff
@@ -1264,15 +1276,28 @@ Subscriber Certificates issued for delegated OCSP responders SHALL have a Validi
...
-
-For all CAs operating under this policy, the computer security functions listed below are required. These functions may be provided by the operating system, or through a combination of operating system, software, and physical safeguards. The CA and its ancillary parts shall include the following functionality:  
+Administrator privileges to all Certificate System components SHALL only be granted to the Administrator trusted role. The CA SHALL enforce multi-factor authentication using a unique credential created by or assigned to the trusted role for all accounts capable of directly causing certificate issuance or authenticating to Certificate Systems. Multifactor authentication SHALL be implemented on all Certificate System components. If multifactor authentication is not supported or is not technically feasible to implement, the following username and password controls, where technically feasible, SHALL be implemented:
+
+- For accounts that are not publicly accessible (accessible only within a Secure or High Security Zone), require passwords have at leaast twelve (12) characters.
```
Comment (lachellel) [inline certificate-policy.md]: Remove the password allowance. 

Even though the network security requirements "allow" it, USG does not and should not.  These would be privileged accounts and USG requires 2FA for privileged accounts everywhere.    :+1:

File: certificate-policy.md [hunk 2]
```diff
@@ -1264,15 +1276,28 @@ Subscriber Certificates issued for delegated OCSP responders SHALL have a Validi
...
+- For accounts that are publicly accessible (accessible from outside a Secure or High Security Zone), require passwrods have the following:
+  - At least twelve (12) charactes,
+  - Be changed every 90 days,
+  - Use a combination of numeric, alphanumeric, and special characters
+  - Are not a dictionary word or on a list of previously disclosed human-generated passwords
+  - Not be the same password used in the previous four passwords,
+  - Implement a lockout for failed access attempts after three (3) failed access attempts
```
Comment (lachellel) [inline certificate-policy.md:1294]: Keep the lockout regardless of authentication mode (not specific to the un/pwd requirements)

My 2 cents.

File: certificate-policy.md [hunk 3]
```diff
@@ -1302,26 +1331,43 @@ The system development controls for all CAs and any Registration Authority funct
...
+- All system and trusted role accounts be reviewed at least every ninety (90) days. Any account that is no longer in use or necessary for operations be deactivated.
+- Implement a process that disables physical and logical access to a Certificate System by either a privileged user or a trusted role within 24 hours upon termination of the individual's employment or contracting relationship with the CA or Delegated Third Party.
+- Change authentication keys and passwords for any account or trusted role ona Certificate System whenever authorization to access the account is changed or revoked.
 
 ### 6.6.3 Life cycle security controls
 No stipulation.
 
 ## 6.7 Network security controls
-The network security controls for all CAs and any Registration Authority functions listed below are required:
+Secure Zones are a physical or logical separation of Certificate Systems while a High Security Zone is a physical area where a private key or cryptographic equipment is stored. Each Zone is protected commensurate with its level of assurance. A High Security Zone may exist within a Secure Zone that is physically or logically separated from other Secure Zones.
```
Comment (lachellel) [inline certificate-policy.md:1352]: @weirdscience 
i pushed a new commit that rewrote 6.7
can you review?

**Review by weirdscience — COMMENTED** [review]
I like it.

File: certificate-policy.md (hunk 1 above)
Comment (weirdscience) [reply certificate-policy.md]: Not all systems support 2FA and for those that don't we should stipulate a password policy. Maybe adding a line about where not technically feasible documenting the level of effort/cost to implementing 2FA? The Fed 2FA policy does allow exceptions.

There's multiple types of privileged accounts and I'm thinking both the OS and CA component level. M of N use passwords even though they are multiparty controls.

File: certificate-policy.md (hunk 2 above)
Comment (weirdscience) [reply certificate-policy.md:1294]: Agree.

File: certificate-policy.md (hunk 3 above)
Comment (weirdscience) [reply certificate-policy.md:1352]: Looks good, a little cleaner and more organized.

File: certificate-policy.md (hunk 1 above)
Comment (lachellel) [reply certificate-policy.md]: Where does the Fed 2FA policy allow exceptions?  (i'm not referring to Certificate Policy but USG policy).

Comment (weirdscience) [reply certificate-policy.md]: It sounds like we should make the language more specific that any trusted role with operating system privileged access are required to have MFA to the operating system. Is the Fed 2FA policy CSIP?

A couple factors to consider with this (assuming one factor is PIV):
1) Issuance time of a PIV. If it take weeks or months to get a PIV, that has real contract impact. Alternative is use a locally issued token until a PIV is issued or use a locally issued token.
2) Availability of the infrastructure. If there was an availability issue in retrieving revocation data, would that impact PIV logon. (example of the PIV logon to the issuing CA system that does not have network connectivity.
3) Offline systems. PIV can be deployed without a domain controller.

Comment (lachellel) [reply certificate-policy.md]: PIV is two factor (but irrelevant to this).  We don't need to specify the hardware token. :+1:

Comment (weirdscience) [reply certificate-policy.md]: Correct, Two factor may be difficult for air-gapped or offline systems. I think defining air-gapped and offline will be helpful as well. Air-gapped meaning no internet connection and physically separate from connected systems while Offline is a air-gapped system only turned on for specific certificate actions. One thing the CAB Forum Policy Working Group attempted to do was define clearly identify which requirements align with what type of CA (differentiating between issuing and root CAs). Multifactor should be a requirement for online systems, but may not be a requirement for offline or air-gapped based on the commensurate physical and logical controls and lowered authentication risk. More factors to consider for MFA on offline / air-gapped systems:

- Additional components that are needed to conduct MFA (i.e. domain controller, authentication servers that may be require an internet connection, etc).
- The ATO or FISMA requirements placed on those additional pieces to do MFA.
- The CA software market is very limited and may not natively support simple MFA.

**Comment by weirdscience** [comment]
Can someone else review these requirements? I need these finalized before I can add in the individual trust store requirements.

**Comment by lachellel** [comment]
:+1: 

lgtm - merging

**Pull request merged**